src/sonic-config-engine/tests/sample_output/py2/l2switch.json

```diff
@@ -0,0 +1,268 @@
+{
+    "DEVICE_METADATA": {"localhost": {"hwsku": "Mellanox-SN2700"}},
```

Comment:
>DEVICE_METADATA": {"localhost": {"hwsku": "Mellanox-SN2700"}}, [](start = 5, length = 62)

I suggest only keep one sample_output for both py2 and py3. You can reorder items in the sonic-cfggen output, then you can compare the output against sample_output order-insensitive in the test script.
It does not make sense to degrade the sample_output to fit your change.

Reply:
> > DEVICE_METADATA": {"localhost": {"hwsku": "Mellanox-SN2700"}}, [](start = 5, length = 62)
> 
> I suggest only keep one sample_output for both py2 and py3. You can reorder items in the sonic-cfggen output, then you can compare the output against sample_output order-insensitive in the test script.
> It does not make sense to degrade the sample_output to fit your change.

I agree with you that one sample_output should be the way forward. This is a long running task and few issues are coming from the templates using loop index. Also, this requires knowledge of how to sort out configuration files in a way that does not spoil its syntax such that we do not end up with passing test cases while element in the files are shuffled. Configuration files span multiple domains and the effort will delay conversion to Python 3.

Also, another point to add here is that we are going to deprecate Python 2 packaging and so the folders that has Python 2 will go away soon.